vendingMachine/src/vendingMachine.js

```diff
@@ -0,0 +1,60 @@
+class VendingMachine {
+  constructor(
+    stock = [
+      { name: "Smarties", code: "A01", quantity: 10, price: 0.6 },
+      { name: "Caramac Bar", code: "A02", quantity: 5, price: 0.6 },
+      { name: "Dairy Milk", code: "A03", quantity: 1, price: 0.65 },
+      { name: "Freddo", code: "A04", quantity: 1, price: 0.25 },
+    ],
+    float = 10.0
+  ) {
+    this.stock = stock;
+    this.float = float;
+  }
+
+  findItem(code) {
+    let item = this.stock.find((item) => {
+      return item.code === code;
+    });
+    return item;
+  }
+
+  calculateChange(itemPrice, money) {
+    let change = (money - itemPrice).toFixed(2);
+    return change;
+  }
+
+  vend(itemCode, money) {
+    let selectedItem = this.findItem(itemCode);
+
+    // is this good practise, setting something to undefined?
```

Comment:
Javascript developers play pretty fast and loose with `undefined`, but yeah it's fine in this case. It's pretty easy to read as you can look just above and see that `undefined` represents the item not being there.